### PR TITLE
Feat/add poap and nft to rebase dapp

### PR DESCRIPTION
**Pull request by krhoda**
Includes PRs #50, #52, #53, and #54 as well as additional changes to repair issues with the Svelte app (the "destiny" operator, `$:` seemed to break, so it's usage has been replaced with `writable` vars). 

As long is it is working, then this is a success. To test, use the CloudFlare link below, if the flows are working, it's ready for merge.

### CODE REVIEW

File: demo/witness/worker/worker.js [hunk 1]
```diff
@@ -233,6 +239,8 @@ function witnessOpts() {
   if (usePoap) {
     o.poap_ownership = {
       api_key: POAP_API_KEY,
+      // TODO: Put in conf obj:
+      challenge_delimiter: "\n\n",
```
Comment (chunningham) [inline demo/witness/worker/worker.js:243]: I am probably confused, was there at some point a PR to do the config object separate from the secrets?

**Review by chunningham — APPROVED** [review]
only minor non-blocking nitpicks, looks nice! very organised for a large pr

File: demo/witness/worker/worker.js (hunk 1 above)
Comment (chunningham) [reply demo/witness/worker/worker.js:243]: ah, #58, not sure how I missed it when looking

File: demo/dapp/src/components/form/WitnessForm.svelte [hunk 2]
```diff
@@ -119,12 +145,17 @@
         } else {
             witnessState.set("signer");
         }
+
         try {
+            console.log("Here?");
             let res = await client.instructions(JSON.stringify({ type }));
             let instruction_res = JSON.parse(res);
+            console.log("Here??");
```
[the remark this answers is not in the document]
Comment (krhoda) [reply demo/dapp/src/components/form/WitnessForm.svelte:153]: Good point, caught that in a later PR

File: demo/dapp/src/components/form/WitnessFormWitness.svelte [hunk 3]
```diff
@@ -15,6 +15,14 @@
     export let post: Function;
     export let back: Function;
     export let advance: Function;
+
+    const needsCopyArea = (t: CredentialType): boolean => {
+        return t !== "email" && t !== "nft_ownership" && t !== "poap_ownership";
+    };
+
+    const needsInput = (t: CredentialType): boolean => {
+        return t === "twitter" || t === "github" || t === "email";
```
[the remark this answers is not in the document]
Comment (krhoda) [reply demo/dapp/src/components/form/WitnessFormWitness.svelte:24]: No, we haven't supported that in Rebase yet.

**Review by anukritidata — APPROVED** [review]
The PR is building and working.

**Pull request merged**